entities/block.coffee

```diff
@@ -9,6 +9,7 @@ class Block
     statement.analyze(localContext) for statement in @statements
 
   optimize: ->
-    # TODO
-
+    @statements = (s.optimize() for s in @statements)
```

Comment:
Someone is :smile: 

It's not a huge deal, it just looked awkward.

Also, it's a nice [duck](https://rachelbythebay.com/w/2013/06/05/duck/).